<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TextEditor::Internal::CompletionSettingsPage</class>
 <widget class="QWidget" name="TextEditor::Internal::CompletionSettingsPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>719</width>
    <height>567</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Behavior</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="1" column="0">
       <widget class="QLabel" name="completionTriggerLabel">
        <property name="text">
         <string>Activate completion:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2" colspan="4">
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>70</width>
          <height>24</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="caseSensitivityLabel">
        <property name="text">
         <string>&amp;Case-sensitivity:</string>
        </property>
        <property name="buddy">
         <cstring>caseSensitivity</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="caseSensitivity">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <item>
         <property name="text">
          <string>Full</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>None</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>First Letter</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="1" column="1" colspan="2">
       <widget class="QComboBox" name="completionTrigger">
        <item>
         <property name="text">
          <string>Manually</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>When Triggered</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Always</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QLabel" name="automaticProposalTimeoutLabel">
        <property name="text">
         <string>Timeout in ms:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="4">
       <widget class="QSpinBox" name="automaticProposalTimeoutSpinBox">
        <property name="maximum">
         <number>500</number>
        </property>
        <property name="singleStep">
         <number>50</number>
        </property>
        <property name="value">
         <number>400</number>
        </property>
       </widget>
      </item>
      <item row="1" column="5">
       <spacer name="horizontalSpacer_3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>24</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="2" column="0">
       <widget class="QCheckBox" name="partiallyComplete">
        <property name="toolTip">
         <string>Inserts the common prefix of available completion items.</string>
        </property>
        <property name="text">
         <string>Autocomplete common &amp;prefix</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QCheckBox" name="autoSplitStrings">
        <property name="toolTip">
         <string>Splits a string into two lines by adding an end quote at the cursor position when you press Enter and a start quote to the next line, before the rest of the string.

In addition, Shift+Enter inserts an escape character at the cursor position and moves the rest of the string to the next line.</string>
        </property>
        <property name="text">
         <string>Automatically split strings</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="autoInsertGroupBox">
     <property name="title">
      <string>&amp;Automatically insert matching characters</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QCheckBox" name="insertBrackets">
        <property name="text">
         <string>Insert opening or closing brackets</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QCheckBox" name="insertQuotes">
        <property name="text">
         <string>Insert closing quote</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QCheckBox" name="surroundBrackets">
        <property name="toolTip">
         <string>When typing a matching bracket and there is a text selection, instead of removing the selection, surrounds it with the corresponding characters.</string>
        </property>
        <property name="text">
         <string>Surround text selection with brackets</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QCheckBox" name="spaceAfterFunctionName">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>Insert &amp;space after function name</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QCheckBox" name="surroundQuotes">
        <property name="toolTip">
         <string>When typing a matching quote and there is a text selection, instead of removing the selection, surrounds it with the corresponding characters.</string>
        </property>
        <property name="text">
         <string>Surround text selection with quotes</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QCheckBox" name="animateAutoComplete">
        <property name="toolTip">
         <string>Show a visual hint when for example a brace or a quote is automatically inserted by the editor.</string>
        </property>
        <property name="text">
         <string>Animate automatically inserted text</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QCheckBox" name="highlightAutoComplete">
        <property name="text">
         <string>Highlight automatically inserted text</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <spacer name="horizontalSpacer_6">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Fixed</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>30</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QCheckBox" name="skipAutoComplete">
          <property name="toolTip">
           <string>Skip automatically inserted character if re-typed manually after completion.</string>
          </property>
          <property name="text">
           <string>Skip automatically inserted character when typing</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="5" column="0">
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <spacer name="horizontalSpacer_7">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Fixed</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>30</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QCheckBox" name="removeAutoComplete">
          <property name="toolTip">
           <string>Remove the automatically inserted character if the trigger is deleted by backspace after the completion.</string>
          </property>
          <property name="text">
           <string>Remove automatically inserted text on backspace</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="docCommentsGroup">
     <property name="title">
      <string>Documentation Comments</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QCheckBox" name="enableDoxygenCheckBox">
        <property name="toolTip">
         <string>Automatically creates a Doxygen comment upon pressing enter after a '/**', '/*!', '//!' or '///'.</string>
        </property>
        <property name="text">
         <string>Enable Doxygen blocks</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <spacer name="horizontalSpacer_5">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Fixed</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>30</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QCheckBox" name="generateBriefCheckBox">
          <property name="toolTip">
           <string>Generates a &lt;i&gt;brief&lt;/i&gt; command with an initial description for the corresponding declaration.</string>
          </property>
          <property name="text">
           <string>Generate brief description</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QCheckBox" name="leadingAsterisksCheckBox">
        <property name="toolTip">
         <string>Adds leading asterisks when continuing C/C++ &quot;/*&quot;, Qt &quot;/*!&quot; and Java &quot;/**&quot; style comments on new lines.</string>
        </property>
        <property name="text">
         <string>Add leading asterisks</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="extraStartEndLines">
        <property name="text">
         <string>Add extra start and end lines</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer_2">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>caseSensitivity</tabstop>
  <tabstop>completionTrigger</tabstop>
  <tabstop>automaticProposalTimeoutSpinBox</tabstop>
  <tabstop>partiallyComplete</tabstop>
  <tabstop>autoSplitStrings</tabstop>
  <tabstop>insertBrackets</tabstop>
  <tabstop>insertQuotes</tabstop>
  <tabstop>surroundBrackets</tabstop>
  <tabstop>surroundQuotes</tabstop>
  <tabstop>spaceAfterFunctionName</tabstop>
  <tabstop>animateAutoComplete</tabstop>
  <tabstop>highlightAutoComplete</tabstop>
  <tabstop>skipAutoComplete</tabstop>
  <tabstop>removeAutoComplete</tabstop>
  <tabstop>enableDoxygenCheckBox</tabstop>
  <tabstop>generateBriefCheckBox</tabstop>
  <tabstop>leadingAsterisksCheckBox</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>enableDoxygenCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>generateBriefCheckBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>216</x>
     <y>411</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>438</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>highlightAutoComplete</sender>
   <signal>toggled(bool)</signal>
   <receiver>skipAutoComplete</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>176</x>
     <y>277</y>
    </hint>
    <hint type="destinationlabel">
     <x>186</x>
     <y>306</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>highlightAutoComplete</sender>
   <signal>toggled(bool)</signal>
   <receiver>removeAutoComplete</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>223</x>
     <y>275</y>
    </hint>
    <hint type="destinationlabel">
     <x>226</x>
     <y>333</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
